otter/test/log/test_cloudfeeds.py

```diff
@@ -211,39 +224,87 @@ def _get_request(self, _type, _id, tenant_id):
+        Given a sequence of functions that take an intent and returns a
+        response (or raises an exception), perform :func:`add_event` and
+        return the result.
         """
         log = object()
         eff = add_event(self.event, 'tid', 'ord', log)
-
-        # effect is to generate UUID
-        self.assertIs(eff.intent.func, uuid.uuid4)
-        eff = resolve_effect(eff, uuid.UUID(int=0))
         uid = '00000000-0000-0000-0000-000000000000'
 
-        # effect scoped on on tenant id
-        self.assertIs(type(eff.intent), TenantScope)
-        self.assertEqual(eff.intent.tenant_id, 'tid')
-
-        # Wrapped effect is retry
-        eff = eff.intent.effect
-        self.assertEqual(
-            eff.intent.should_retry,
-            ShouldDelayAndRetry(can_retry=retry_times(5),
-                                next_interval=exponential_backoff_interval(2)))
-
-        # effect wrapped in retry is ServiceRequest
-        eff = eff.intent.effect
-        self.assertEqual(
-            eff,
-            service_request(
-                ServiceType.CLOUD_FEEDS, 'POST', 'autoscale/events',
-                headers={
-                    'content-type': ['application/vnd.rackspace.atom+json']},
-                data=self._get_request('INFO', uid, 'tid'), log=log,
-                success_pred=has_code(201)))
+        svrq = service_request(
+            ServiceType.CLOUD_FEEDS, 'POST', 'autoscale/events',
+            headers={
+                'content-type': ['application/vnd.rackspace.atom+json']},
+            data=self._get_request('INFO', uid, 'tid'), log=log,
+            success_pred=has_code(201),
+            json_response=False).intent
+
+        def perform_retry_without_delay(retry_intention):
+            def should_retry(exc_info):
+                exc_type, exc_value, exc_traceback = exc_info
+                failure = Failure(exc_value, exc_type, exc_traceback)
+                return Effect(Constant(
+                    retry_intention.should_retry.can_retry(failure)))
+
+            seq = [(svrq, resp) for resp in response_sequence]
+
+            new_retry_intent = Effect(Retry(effect=retry_intention.effect,
+                                            should_retry=should_retry))
+
+            return perform_sequence(
+                seq, new_retry_intent,
+                test_dispatcher(TypeDispatcher({Retry: perform_retry})))
+
+        seq = [
+            (Func(uuid.uuid4), lambda _: uid),
+            (TenantScope(mock.ANY, 'tid'), nested_sequence([
+                (Retry(effect=mock.ANY, should_retry=ShouldDelayAndRetry(
+                    can_retry=mock.ANY,
+                    next_interval=exponential_backoff_interval(2))),
+                 # we actually want to perform the effect multiple times so
+                 # we can test retrying
+                 perform_retry_without_delay)
+                ]))
+        ]
+
+        return perform_sequence(seq, eff)
```

Comment:
I haven't looked at this test too closely, just this comment thread, but one option would be to use the real `ServiceRequest` performer (`concretize_service_request`) here, which would cause the pure_http.Request to be run (meaning you could put it in the `SequenceDispatcher` list right after the `ServiceRequest`)

that may be tedious/ugly, so other options may be better